README.md

```diff
@@ -2,32 +2,42 @@
 
 ## Introduction to Substreams
 
-Substreams is a powerful blockchain indexing technology, developed for The Graph Network.
+Substreams is a robust blockchain indexing technology developed for The Graph Network.
 
-Substreams enables developers to write Rust modules, composing data streams alongside the community, and provides extremely high performance indexing by virtue of parallelization, in a streaming-first fashion.
-Substreams has all the benefits of StreamingFast Firehose, like low-cost caching and archiving of blockchain data, high throughput processing, and cursor-based reorgs handling.
+Substreams empowers developers to create Rust modules, facilitating the composition of data streams in collaboration with the community. It offers exceptionally high-performance indexing through parallelization, all while embracing a streaming-first approach. Substreams inherits the advantages of StreamingFast Firehose, including cost-effective caching and archiving of blockchain data, high throughput processing, and robust handling of cursor-based reorganizations.
 
 ## Documentation
 
-Full documentation for installing, running and working with Substreams is available at: https://substreams.streamingfast.io.
+For comprehensive documentation on installing, running, and working with Substreams, please visit: [Substreams Documentation](https://substreams.streamingfast.io).
 
-## Streaming orders and trade data from the Verified Network
+## Streaming Orders and Trade Data from the Verified Network
 
-The modules in this repo stream subscription/pricing data from primary issue pools, and trade/pricing data from secondary issue pools.
+This repository contains modules that stream subscription/pricing data from primary issue pools and trade/pricing data from secondary issue pools. There are two modules available: PrimaryPool and SecondaryPool. Substreams stores the data in key/value stores, making it accessible via gRPC. 
+More details on modules can be found [Primary](./modules/PrimaryPool/) and [Secondry](./modules/SecondaryPool/)
+More details on client interfaces can be found [here](https://github.com/streamingfast/substreams-sink-kv/tree/develop/examples/generic-service) along with an illustrative example.
```

Comment:
@yaksheshgupta better to remove, I do not think this generic service helps anyone. Our example app is a good enough example.